Pod/Extensions/SSO/SKYAuthContainer+SSO.m

```diff
@@ -0,0 +1,200 @@
+           dictionaryArguments:@{@"access_token" : accessToken}
+             completionHandler:^(NSDictionary *result, NSError *error) {
+                 [weakSelf _handleLoginOAuthResult:result
+                                             error:error
+                                 completionHandler:completionHandler];
+             }];
+}
+
+- (void)linkOAuthProvider:(NSString *_Nonnull)providerID
+              accessToken:(NSString *_Nonnull)accessToken
+        completionHandler:(void (^_Nullable)(NSError *_Nullable))completionHandler
+{
+    [self.container callLambda:[self _getAuthURLWithAction:SKYOAuthActionLink]
+           dictionaryArguments:@{@"access_token" : accessToken}
+             completionHandler:^(NSDictionary *result, NSError *error) {
+                 completionHandler(error);
+             }];
+}
+
+- (void)_oauthFlowWithProvider:(NSString *)providerID
+                       options:(NSDictionary *)options
+                        action:(SKYOAuthActionType)action
+             completionHandler:(SKYWebOAuthCompletion)completionHandler
+{
+    NSError *validateError = [self _validateGetAuthURLParams:options];
+    if (validateError) {
+        completionHandler(nil, validateError);
+        return;
+    }
+    NSDictionary *params = [self _genAuthURLParams:options];
+    NSString *urlFormat = [self _getAuthURLWithAction:action];
+    NSURL *callbackURL = [self _genCallbackURL:options[@"scheme"]];
+
+    [[self container] callLambda:[NSString stringWithFormat:urlFormat, providerID]
+             dictionaryArguments:params
+               completionHandler:^(NSDictionary *result, NSError *error) {
+                   if (error != nil) {
+                       completionHandler(nil, error);
+                       return;
+                   }
+                   [[SKYWebOAuth shared] startOAuthFlow:result[@"auth_url"]
+                                            callbackURL:callbackURL
+                                      completionHandler:completionHandler];
+               }];
+}
+
+- (void)_handleLoginOAuthResult:(NSDictionary *)result
+                          error:(NSError *)error
+              completionHandler:(SKYContainerUserOperationActionCompletion)completionHandler
+{
+    NSError *loginError = error;
+    SKYRecord *user = nil;
+    SKYAccessToken *accessToken = nil;
+    if (!loginError) {
+        NSDictionary *response = result[@"result"];
+        NSDictionary *profile = response[@"profile"];
+        NSString *recordID = profile[@"_id"];
+        if ([recordID hasPrefix:@"user/"] && response[@"access_token"]) {
+            user = [[SKYRecordDeserializer deserializer] recordWithDictionary:profile];
+            accessToken = [[SKYAccessToken alloc] initWithTokenString:response[@"access_token"]];
+
+            [self updateWithUser:user accessToken:accessToken];
+            // register device when login and signup success
+            [self.container.push registerDeviceCompletionHandler:nil];
```

Comment:
Let’s put `registerDeviceCompletionHandler:` inside `updateWithUser:accessToken:`? Basically we want to register when the user changes to another user, and we unregister when the user changes to `nil`.